hash_practice/exercises.py

```diff
@@ -2,18 +2,44 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)
+        Space Complexity: O(n)
     """
-    pass
+    string_dict = {}
+    for string in strings:
+        key = ''.join(sorted(string))
+
+        if key in string_dict.keys():
+            string_dict[key].append(string)
+        else:
+            string_dict[key] = [string]
+        
+    return list(string_dict.values())
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n) + O(n log n)??

```

Comment:
This is almost right. Usually when doing big-o notation, everything goes in one "O" so your answer would be better expressed as O(n + n * log(n)). However, in that case, as the `n * log(n)` part will increase in size quicker than the `n` part (like when `n^2` or `n * n` dwarfs `n`), we can just only keep that part to reduce down to O(n * log(n)).

Your space complexity and implementation are fine, though.